packages/api/src/target-group/target-groups.service.ts

```diff
@@ -66,53 +67,62 @@ export class TargetGroupsService {
+                    offset += contacts.length;
+
+                    const contactsInContactList: BrevoContactInterface[] = [];
+                    const contactsNotInContactList: BrevoContactInterface[] = [];
+
+                    for (const contact of contacts) {
+                        const contactIsInTargetGroupByFilters = this.checkIfContactIsInTargetGroup(contact.attributes, filters);
+
+                        const manuallyAssignedTargetGroup = targetGroup.assignedContactsTargetGroupBrevoId;
+                        const contactIsManuallyAssignedToTargetGroup = manuallyAssignedTargetGroup
+                            ? contact.listIds.includes(manuallyAssignedTargetGroup)
+                            : false;
+
+                        if (contactIsInTargetGroupByFilters || contactIsManuallyAssignedToTargetGroup) {
+                            contactsInContactList.push(contact);
+                        } else {
+                            contactsNotInContactList.push(contact);
+                        }
+                    }
 
-                if (contactIsInTargetGroupByFilters || contactIsManuallyAssignedToTargetGroup) {
-                    contactsInContactList.push(contact);
-                } else {
-                    contactsNotInContactList.push(contact);
+                    if (contactsInContactList.length > 0) {
+                        await this.brevoApiContactsService.updateMultipleContacts(
+                            contactsInContactList.map((contact) => ({ id: contact.id, listIds: [targetGroup.brevoId] })),
+                            scope,
+                        );
+                    }
+                    if (contactsNotInContactList.length > 0) {
+                        await this.brevoApiContactsService.updateMultipleContacts(
+                            contactsNotInContactList.map((contact) => ({ id: contact.id, unlinkListIds: [targetGroup.brevoId] })),
+                            scope,
+                        );
+                    }
+                } catch (error) {
+                    handleBrevoError(error);
                 }
-            }
+            } while (offset < totalCount);
 
-            if (contactsInContactList.length > 0) {
-                await this.brevoApiContactsService.updateMultipleContacts(
-                    contactsInContactList.map((contact) => ({ id: contact.id, listIds: [targetGroup.brevoId] })),
-                    scope,
-                );
-            }
-            if (contactsNotInContactList.length > 0) {
-                await this.brevoApiContactsService.updateMultipleContacts(
-                    contactsNotInContactList.map((contact) => ({ id: contact.id, unlinkListIds: [targetGroup.brevoId] })),
-                    scope,
-                );
-            }
-        } while (offset < totalCount);
-
-        return true;
+            return true;
+        } catch (error) {
+            handleBrevoError(error);
+            throw error;
```

Comment:
Why is this needed? handleBrevo throws an error in any case, so this line can never be reached right?

Reply:
88bcc36438b1f9ebc033777055f07ec9f2b46abe